File: backend/parking-domain/src/main/java/com/chatchatabc/parking/impl/infra/service/UtilServiceImpl.java
package com.chatchatabc.parking.impl.infra.service;

import com.chatchatabc.parking.infra.service.UtilService;
import org.springframework.stereotype.Service;

import java.util.Locale;
import java.util.Random;

@Service
public class UtilServiceImpl implements UtilService {
    private final Random random = new Random();

    /**
     * Generate OTP
     *
     * @return OTP
     */
    @Override
    public String generateOTP() {
        int randomNumber = random.nextInt(1000000);
        return String.format(Locale.ENGLISH, "%06d", randomNumber);
    }
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/parking/impl/infra/service/UtilServiceImpl.kt
package com.chatchatabc.parking.impl.infra.service

import com.chatchatabc.parking.infra.service.UtilService
import org.springframework.stereotype.Service

@Service
class UtilServiceImpl : UtilService {
    /**
     * Generate OTP
     */
    override fun generateOTP(): String {
        return (0..999999).random().toString().padStart(6, '0')
    }
}